### DISCUSSION TITLE
Pi4J-v2 with sound cards

### DISCUSSION THREAD

**Discussion by Grob98** [Pi4J/pi4j#174] [General]
Hi,

I have an sound card hat connected to the raspberry pi. When I register a output gpio (not used by the sound card) with pi4j-v2 the sound card stops working or the sound is distorted. It works with pi4j-v1. 

Can anybody help or do you have similar problems?

**Comment by FDelporte** [comment]
Just an idea... maybe one of the loaded modules is causing some data interference on this line?

Are you using this context?

```
var pi4j = Pi4J.newAutoContext();
```

Could you maybe switch to a self-defined context and try to add one by one provider and see if the sound problem is linked to one of them specifically?

```
var piGpio = PiGpio.newNativeInstance();
var pi4j = Pi4J.newContextBuilder()
    .noAutoDetect()
    .add(
        PiGpioDigitalInputProvider.newInstance(piGpio),
        PiGpioDigitalOutputProvider.newInstance(piGpio),
        PiGpioPwmProvider.newInstance(piGpio),
        PiGpioI2CProvider.newInstance(piGpio),
        PiGpioSerialProvider.newInstance(piGpio),
        PiGpioSpiProvider.newInstance(piGpio)
    )
    .build();
```

**Reply by Grob98** [reply]
Sorry for my late reply. I tried it again with a self-defined context:

The problem does not occur when creating the context. It occurs when creating a `DigitalOutput` or `DigitalInput`.

**Comment by hagego** [comment]
I just moved my application from V1.4 to V2 and have the same problem, my HiFiBerry is only producing distorted sound. Seems the answer can be found here:
https://abyz.me.uk/rpi/pigpio/faq.html#Sound_isnt_working

It would be great if gpioCfgClock() can be made accessible in PI4J V2

See also https://github.com/Pi4J/pi4j-v2/discussions/99

BTW, I also changed my code already to a self defined context with only the Providers I need (I2C, SPI, Digital in/out), not loading PWM, but it makes no difference
